NU1903 error when run dotnet build

**Summary**

When I try to build a solution, I get errors NU1903

**Details**

```bash
> dotnet build
    .\sdk-dotnet\tests\Temporalio.Tests\Temporalio.Tests.csproj : error NU1903: Warning As Error: Package 'System.Net.Http' 4.3.0 has a known high severity vulnerability, https://github.com/advisories/GHSA-7jgj-8wvc-jh57
    .\sdk-dotnet\src\Temporalio.Extensions.Hosting\Temporalio.Extensions.Hosting.csproj : error NU1903: Warning As Error: Package 'System.Text.Json' 7.0.0 has a known high severity vulnerability, https://github.com/advisories/GHSA-hh2w-p6rv-4g7w
    .\sdk-dotnet\tests\Temporalio.Tests\Temporalio.Tests.csproj : error NU1903: Warning As Error: Package 'System.Text.Json' 8.0.0 has a known high severity vulnerability, https://github.com/advisories/GHSA-hh2w-p6rv-4g7w
    .\sdk-dotnet\tests\Temporalio.Tests\Temporalio.Tests.csproj : error NU1903: Warning As Error: Package 'System.Text.RegularExpressions' 4.3.0 has a known high severity vulnerability, https://github.com/advisories/GHSA-cmhx-cq75-c4mj
```

Restore failed with 4 error(s) in 0.7s

@cretz hello, sorry for delay, I reproduced it :) https://github.com/rstm-sf/sdk-dotnet/actions/runs/11425577349/job/31787420591?pr=1

```bash
Run dotnet build
  Determining projects to restore...
/home/runner/work/sdk-dotnet/sdk-dotnet/src/Temporalio/Temporalio.csproj : error NU1903: Warning As Error: Package 'System.Text.Json' 8.0.4 has a known high severity vulnerability, https://github.com/advisories/GHSA-8g4q-xg66-9fp4 [/home/runner/work/sdk-dotnet/sdk-dotnet/Temporalio.sln]
  Restored /home/runner/work/sdk-dotnet/sdk-dotnet/src/Temporalio.Extensions.OpenTelemetry/Temporalio.Extensions.OpenTelemetry.csproj (in 3.07 sec).
  Restored /home/runner/work/sdk-dotnet/sdk-dotnet/src/Temporalio.Extensions.Hosting/Temporalio.Extensions.Hosting.csproj (in 3.09 sec).
  Failed to restore /home/runner/work/sdk-dotnet/sdk-dotnet/src/Temporalio/Temporalio.csproj (in 3.07 sec).
  Restored /home/runner/work/sdk-dotnet/sdk-dotnet/src/Temporalio.Extensions.DiagnosticSource/Temporalio.Extensions.DiagnosticSource.csproj (in 4 ms).
  Restored /home/runner/work/sdk-dotnet/sdk-dotnet/tests/Temporalio.Tests/Temporalio.Tests.csproj (in 6.95 sec).

Build FAILED.

/home/runner/work/sdk-dotnet/sdk-dotnet/src/Temporalio/Temporalio.csproj : error NU1903: Warning As Error: Package 'System.Text.Json' 8.0.4 has a known high severity vulnerability, https://github.com/advisories/GHSA-8g4q-xg66-9fp4 [/home/runner/work/sdk-dotnet/sdk-dotnet/Temporalio.sln]
    0 Warning(s)
    1 Error(s)

Time Elapsed 00:00:10.98
Error: Process completed with exit code 1.
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -15,6 +15,10 @@ jobs:
         include:
           - os: windows-latest
             checkTarget: true
+            # We want to use one version for most things and this latest for
+            # others. This is an ok way to test both without requiring a
+            # separate matrix permutation.
+            dotNetVersionOverride: 8.x
           - os: ubuntu-latest
             docsTarget: true
             cloudTestTarget: true
@@ -44,7 +48,7 @@ jobs:
         with:
           # Specific .NET version required because GitHub macos ARM image has
           # bad pre-installed .NET version
-          dotnet-version: 6.x
+          dotnet-version: ${{ matrix.dotNetVersionOverride || 6 }}
 
       - name: Install protoc
         uses: arduino/setup-protoc@v3

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -48,7 +48,7 @@ jobs:
         with:
           # Specific .NET version required because GitHub macos ARM image has
           # bad pre-installed .NET version
-          dotnet-version: ${{ matrix.dotNetVersionOverride || 6 }}
+          dotnet-version: ${{ matrix.dotNetVersionOverride || '6.x' }}
 
       - name: Install protoc
         uses: arduino/setup-protoc@v3

--- a/src/Temporalio/Temporalio.csproj
+++ b/src/Temporalio/Temporalio.csproj
@@ -20,7 +20,7 @@
 
   <ItemGroup Condition="'$(TargetFramework)' == 'net462' Or '$(TargetFramework)' == 'netstandard2.0'">
     <PackageReference Include="Microsoft.Bcl.HashCode" Version="1.1.1" />
-    <PackageReference Include="System.Text.Json" Version="8.0.4" />
+    <PackageReference Include="System.Text.Json" Version="8.0.5" />
   </ItemGroup>
 
   <!-- Allow tests to use internals -->

